File: data/src/main/java/home/oleg/placesnearme/data/mapper/VenueMapper.java
package home.oleg.placesnearme.data.mapper;

import java.util.ArrayList;
import java.util.List;

import home.oleg.placenearme.models.Location;
import home.oleg.placenearme.models.Venue;
import home.oleg.placesnearme.network.models.VenueDto;

/**
 * Created by Oleg Sheliakin on 25.09.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
public final class VenueMapper {

    private VenueMapper() {
    }

    public static Venue map(VenueDto venue) {
        Venue result = new Venue();

        for (home.oleg.placesnearme.network.models.Category category : venue.getCategories()) {
            if (category.getPrimary()) {
                result.setCategory(CategoryMapper.INSTANCE.map(category));
            }
        }

        result.setId(venue.getId());

        Location location = LocationMapper.map(venue.getLocation());

        result.setLocation(location);
        result.setName(venue.getName());

        return result;
    }

    public static List<Venue> map(List<VenueDto> venues) {
        List<Venue> result = new ArrayList<>();

        for (VenueDto venue : venues) {
            result.add(map(venue));
        }

        return result;
    }

}

File: data/src/main/java/home/oleg/placesnearme/data/mapper/VenueMapper.kt
package home.oleg.placesnearme.data.mapper

import java.util.ArrayList

import home.oleg.placenearme.models.Location
import home.oleg.placenearme.models.Venue
import home.oleg.placesnearme.network.models.VenueDto

/**
 * Created by Oleg Sheliakin on 25.09.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
object VenueMapper {

    fun map(venue: VenueDto) = Venue(
            name = venue.name,
            category = venue.categories.first { it.primary }.let { CategoryMapper.map(it) },
            id = venue.id,
            location = LocationMapper.map(venue.location)
    )

    fun map(venues: List<VenueDto>): List<Venue> {
        val result = ArrayList<Venue>()

        for (venue in venues) {
            result.add(map(venue))
        }

        return result
    }

}
